### DISCUSSION TITLE
Can resources be published in WireUI v2?

### DISCUSSION THREAD

**Discussion by JohnMSykes** [wireui/wireui#912] [Q&A]
Hi - We have extended WireUI 1.x by adding custom icons to our locally published WireUI resources. These resources were originally created using this publish command:

`php artisan vendor:publish --tag='wireui.resources'`

This is an example of a custom icon we added to the original v1 Heroicon set:

`emoji-neutral.blade.php`

```
<svg {{ $attributes }} xmlns="http://www.w3.org/2000/svg" fill="currentColor" viewBox="0 0 24 24" stroke="currentColor">
    <path stroke-linecap="round" stroke-linejoin="round" stroke-width="1" d="M12,22 C6.4771525,22 2,17.5228475 2,12 C2,6.4771525 6.4771525,2 12,2 C17.5228475,2 22,6.4771525 22,12 C22,17.5228475 17.5228475,22 12,22 Z M12,21 C16.9705627,21 21,16.9705627 21,12 C21,7.02943725 16.9705627,3 12,3 C7.02943725,3 3,7.02943725 3,12 C3,16.9705627 7.02943725,21 12,21 Z M7.5,15 C7.22385763,15 7,14.7761424 7,14.5 C7,14.2238576 7.22385763,14 7.5,14 L16.5,14 C16.7761424,14 17,14.2238576 17,14.5 C17,14.7761424 16.7761424,15 16.5,15 L7.5,15 Z M9,11 C8.44771525,11 8,10.5522847 8,10 C8,9.44771525 8.44771525,9 9,9 C9.55228475,9 10,9.44771525 10,10 C10,10.5522847 9.55228475,11 9,11 Z M15,11 C14.4477153,11 14,10.5522847 14,10 C14,9.44771525 14.4477153,9 15,9 C15.5522847,9 16,9.44771525 16,10 C16,10.5522847 15.5522847,11 15,11 Z" />
</svg>
```

Today, we started the process to upgrade to Laravel 11 using Laravel Shift - As part of that process/composer updates, we received the WireUI 2.0 release (Congratulations, BTW - We're huge fans). I know we will need to publish the WireUI config due to our use of aliases (Jetstream uses `<x-name>` and we had to change to `<x-wui-name>`....) and we wanted to ask if the ability to publish resources still existed - It's conspicuously absent from the docs.

**Comment by ph7jack** [comment]
Hi @JohnMSykes, the new release doesn't bring a views publishing option. 
The option to publish was good to customize but bad for updating the new code changes, and caused some kind of issues to many developers that have published the views and were not updating them.

In the next weeks, we will have a custom way to use custom icons, you will be able to set the default icon package, or specify the icons package when using any component

**Reply by JohnMSykes** [reply]
Thank you, @PH7-Jack. We were probably in the cohort that wasn't updating enough - we did not implement any post-update Composer actions to keep the publishable assets up to date. Mostly, our changes were as described above, i.e., adding content/icons vs modifying existing.

We sincerely appreciate your and the contributors' efforts - WireUI is a big part of our UI at https://domestiq.net. Thank you, all 🙏🏻.

**Reply by joaopalopes24** [reply]
@JohnMSykes We will improve this part of icons a lot in v2.

**Reply by JohnMSykes** [reply]
Thank you, @joaopalopes24! I was wondering if there was a way to safely "extend" this package in the way that we have done, e.g., to define a safe place for **additions** that could be supported by the underlying package. It would be great to be able to publish just the empty directory structure of the resources and allow uniquely named blade resources to be sourced, i.e.:

Does file/resource exist in vendor resource?
- Yes. Publish/present as normal.
- No. Is this file/resource available in local published resource?
- - Yes. Publish/present as you would for a vendor resource.
- - No. Throw error.

My general thinking is that such a scheme would put all of the responsibility for self-published resources on the user and would maintain the integrity of WireUI-provided resources. 

This is how we used the custom icon I shared in the original post - Of course, all of the other icons and buttons are straight from the Heroicon v1 set that came with WireUI 1.x.x:

![image](https://github.com/wireui/wireui/assets/14906331/77ae8d4d-e6d7-4741-b206-d6515452df09)

Again, we are grateful for the WireUI Team's efforts - It's helped us to build a UX that's playful, intuitive, and effective!

**Comment by madushancs** [comment]
Hi. is posible to update text color in Number component. 
![image](https://github.com/wireui/wireui/assets/45415384/5085bd8a-47fd-4937-abe8-3f3d431e276e)


```
<div x-data="{ count: 0 }" class="flex items-center text-black gap-x-3" style="color: black !important">
 <x-button x-hold.click.repeat.200ms="count--" icon="minus"  class="text-black"
                                                wire:click="updateCartItem({{ $id }}, 'minus')" />

<span class="bg-teal-600 text-white px-5 py-1.5 rounded-lg">
                                                {{$item->get('quantity')}}
</span>

<x-button x-hold.click.repeat.200ms="count++" icon="plus"
                                                wire:click="updateCartItem({{ $id }}, 'plus')" />
</div>
```

**Comment by nastoychev** [comment]
I believe it would be beneficial if we could style input labels to ensure they align with our design guidelines. This would help maintain a consistent and polished appearance across our UI.

I believe it would be a good idea to export the design into a configuration file, allowing the library to remain headless or with a highly flexible UI. This approach would give us more control over styling and ensure that the library can easily adapt to different design systems.

Example: 

```php
return [
    'components' => [
        'input' => [
            // Default properties for input components
            'default' => [
                'labelClass' => 'block mb-2 text-sm font-medium text-gray-900 dark:text-gray-300',
                'inputClass' => 'bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-rose-500 focus:border-rose-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white',
                'errorClass' => 'mt-2 text-sm text-red-600 dark:text-red-400',
                'hintClass'  => 'mt-2 text-sm text-gray-500 dark:text-gray-400',
            ],
        ],
    ],
];
```

For now, my current approach would be to fork the branch and override the class styles as needed and to handle the versioning as same as your standards.

**Comment by emiliosh** [comment]
Any info about how to add custom icons?  Thanks in advance.
